packages/frontend/src/pages/JoinOrganization.tsx

```diff
@@ -75,55 +74,82 @@ export const JoinOrganizationPage: FC = () => {
-                        disabled={disabled}
-                        loading={isCreatingOrg}
-                        onClick={() => createOrg({ name: '' })}
+            <Stack w={400} mt="4xl">
+                <Image
+                    src={LightdashLogo}
+                    alt="lightdash logo"
+                    width={130}
+                    mx="auto"
+                    my="lg"
+                />
+                <Card p="xl" radius="xs" withBorder shadow="xs">
+                    <Stack justify="center" spacing="md" mb="xs">
+                        <Title order={3} ta="center">
+                            Join a workspace
+                        </Title>
+                        <Text color="gray.6" ta="center">
+                            The workspaces below are open to anyone with a{' '}
+                            <b>@{getEmailDomain(user.data?.email || '')}</b>{' '}
+                            domain
+                        </Text>
+                        {allowedOrgs?.map((org) => (
+                            <Card key={org.organizationUuid} withBorder>
+                                <Group position="apart">
+                                    <Group spacing="md">
+                                        <Avatar
+                                            size="md"
+                                            radius="xl"
+                                            color="gray.6"
+                                        >
+                                            {org.name[0]?.toUpperCase()}
+                                        </Avatar>
+                                        <Stack spacing="two">
+                                            <Text truncate fw={600}>
+                                                {org.name}
+                                            </Text>
+                                            <Text fz="xs" c="gray">
+                                                {org.membersCount} members
+                                            </Text>
+                                        </Stack>
+                                    </Group>
+                                    <Button
+                                        onClick={() =>
+                                            joinOrg(org.organizationUuid)
+                                        }
+                                        loading={isJoiningOrg}
+                                    >
+                                        Join
+                                    </Button>
+                                </Group>
+                            </Card>
+                        ))}
+                    </Stack>
+                </Card>
+                <Text ta="center" px="xs">
+                    <Anchor
+                        color={disabled ? 'gray.6' : ''}
+                        onClick={() =>
+                            !disabled ? createOrg({ name: '' }) : null
+                        }
```

Comment:
since this is an anchor and does not actually functions as an anchor you can use polymorphic button `<Button component={Anchor}` (or vice versa, I keep forgetting). It will have a disabled state